macOS/debugger.cpp

```diff
@@ -68,12 +68,37 @@ vm_prot_t Debugger::MacOSProtectionFlags(MemoryProtection memory_protection) {
   }
 }
 
+void Debugger::FreeShare(void *address, size_t size) {
+  if (size == 0) {
+    WARN("FreeShare is called with size == 0\n");
+    return;
+  }
+
+  mach_port_t shm_port = share_mem_map[(mach_vm_address_t)address];
```

Comment:
Shouldn't share_mem_port be used here instead of share_mem_map. Note that later I proposed merging these two maps so it won't matter in that case, but noting for completeness.

Reply:
Sorry for this silly bug, I might be too sleepy lol